src/snowflake/connector/file_transfer_agent.py

```diff
@@ -388,59 +398,73 @@ def _transfer_accelerate_config(self):
-                idx + self._parallel <= len_file_metas else \
-                len_file_metas
-
-            logger.debug(f'uploading files idx: {idx + 1}/{end_of_idx}')
-
-            target_meta = file_metas[idx:end_of_idx]
-            while True:
-                pool = ThreadPoolExecutor(len(target_meta))
-                results: List['SnowflakeFileMeta'] = list(pool.map(
-                    SnowflakeFileTransferAgent.upload_one_file,
-                    target_meta))
-                pool.shutdown()
-
-                # need renew AWS token?
-                retry_meta: List['SnowflakeFileMeta'] = []
-                for result_meta in results:
-                    if result_meta.result_status in [
-                        ResultStatus.RENEW_TOKEN,
-                        ResultStatus.RENEW_PRESIGNED_URL
-                    ]:
-                        retry_meta.append(result_meta)
-                    else:
-                        self._results.append(result_meta)
-
-                if len(retry_meta) == 0:
-                    # no new AWS token is required
-                    break
-                if any([result_meta.result_status == ResultStatus.RENEW_TOKEN
-                        for result_meta in results]):
-                    client = self.renew_expired_client()
-                    for result_meta in retry_meta:
-                        result_meta.client = client
-                    if end_of_idx < len_file_metas:
-                        for idx0 in range(idx + self._parallel, len_file_metas):
-                            file_metas[idx0].client = client
-                if any([result_meta.result_status == ResultStatus.RENEW_PRESIGNED_URL
-                        for result_meta in results]):
+        qtask = Queue()
+        for meta in file_metas:
+            qtask.put(meta)
+
+        retry_round = 0
+        while not qtask.empty():
+            len_file_metas = qtask.qsize()
+            thread_number = len_file_metas if len_file_metas < self._parallel else self._parallel
+            pool = ThreadPoolExecutor(thread_number)
+            qnext = Queue()
+            triggers = list()
+            pool.submit(SnowflakeFileTransferAgent._upload_files_in_queue_thread, qtask, qnext, self._results, triggers)
+            pool.shutdown()
+            if qnext.empty():
+                break
+            if len(triggers) > 0:
+                result_meta = triggers.pop()
+                if result_meta.result_status == ResultStatus.RENEW_PRESIGNED_URL:
                     self._update_file_metas_with_presigned_url()
-                target_meta = retry_meta
 
-            if end_of_idx == len_file_metas:
-                break
-            idx += self._parallel
+            while not qnext.empty():
+                qtask.put(qnext.get())
+            retry_round = retry_round + 1
+            logger.debug("retry {} round for {} items".format(retry_round, qnext.qsize()))
```

Comment:
I don't think that I have told you about f-strings yet, they are kinda cool.
This is not a hard requirement, but I want you  to know what this would look like as an f-string
```suggestion
            logger.debug(f"retry {retry_round} round for {qnext.qsize()} items")
```